src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -1140,7 +1140,11 @@ public override Task<ImplementationResult> ImplementationAsync(DelegatedPosition
             request,
             cancellationToken).ConfigureAwait(false);
 
-        if (response is null)
+        // Sometimes web tools will respond with a non-null response, of an array of diagnostics with a single element
+        // in it, but the actual diagnostics (and every other property) in that element are null. This confuses VS, and
+        // means we don't get squiggles for C# diagnostics, unless there are also Html diagnostics.
+        // This check ensures we catch that, and just return an empty result for Html diagnostics.
+        if (response?.Response is null or [{ Diagnostics: null }, ..])
```

Comment:
If this is too fancy and patterny for people, let me know and I'll un-C#-11-ify it

Reply:
It's pretty pattern-y, but not too hard to parse out if you look up the syntax. I'm cool with it.

Reply:
Cool. Without the comment it's borderline even for me :)

Reply:
Fine by me.